app/helpers/monthly_statistics_export_helper.rb

```diff
@@ -0,0 +1,16 @@
+module MonthlyStatisticsExportHelper
+  def merge_rows_and_totals(data)
+    hash = data[:rows].first.drop(1)
+    totals = data[:column_totals]
+
+    data[:rows] + [{ data[:rows].first.keys.first => 'Total' }.merge!(merge_totals(hash, totals).to_h)]
+  end
+
+private
+
+  def merge_totals(hash, totals)
+    hash.map do |key, _value|
+      [key, totals[hash.find_index { |k, _v| k == key }]]
+    end
+  end
+end
```

Comment:
I found this a bit hard to follow. I had a go at simplifying it and I'm not sure I've made it any clearer, just shorter:

```
module MonthlyStatisticsExportHelper
  def merge_rows_and_totals(data)
    column_names = data[:rows].first.drop(1).map(&:first)
    totals = data[:column_totals]

    data[:rows] + [{ data[:rows].first.keys.first => 'Total' }.merge!(column_names.zip(totals).to_h)]
  end
end
```

Reply:
Yea i think this is def better. Thanks Steve!

TIL about zip